### PR TITLE
Virtual kubelet: do not stop reflection for temporary namespace errors

### LINKED ISSUE #1030: Offloaded pods repatriated after temporary glitch
#### What happened:

A temporary glitch caused all pods offloaded in a given namespace to be locally repatriated. Specifically, the `liqo-controller-manager` failed to contact the remote cluster during a periodic refresh of the namespace mappings (due to the capsule webhook not responding), and marked the corresponding entry in the `namespacemap` as Failed. This, in turn, caused the virtual kubelet to interrupt the reflection, and mark all offloaded pods as `Failed` (`OffloadingAborted`). Immediately, the entire process was re-executed and completed correctly, leading to the restart of the reflection process.

#### What you expected to happen:

Temporary errors should not lead to the repatriation of all resources, but should be addressed with controlled retries. The decision of the repatriation should be up to the virtual kubelet, according to its own timers (by default pods are removed from a `NotReady` node after 300 seconds.

#### How to reproduce it (as minimally and precisely as possible):

1. Install liqo on two clusters
2. Enable a namespace for offloading, and offload some pods
3. Cause the capsule webhook to not respond and observe the behavior during a periodic resync of the namespace controller

#### Environment:
- Liqo version: 0.3.2-beta.2
- Kubernetes version (use `kubectl version`): 1.21
- Cloud provider or hardware configuration: on-premise (kubeadm) 
- Network plugin and version: calico
- Install tools: liqoctl
- Others:

### PR DESCRIPTION
**Pull request by giorio94**
# Description

This PR modifies the virtual kubelet namespace start/stop logic, to prevent pods from being repatriated in case of temporary glitches which cause the NamespaceMap entry to be marked in CreationLoopBackoff state. Specifically, now this transition is ignored, assuming the normal functioning will be later recovered.

Fixes #1030 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Manual testing
- [x] Unit testing (new + existing)
- [x] E2E testing

### CODE REVIEW

File: pkg/virtualKubelet/reflection/namespacemap/namespace.go [hunk 1]
```diff
@@ -104,7 +108,11 @@ func (nh *Handler) onUpdateNamespaceMap(oldObj, newObj interface{}) {
 	for localNs, oldRemoteNamespaceStatus := range oldNamespaceMap.Status.CurrentMapping {
 		newRemoteNamespaceStatus, newRemoteNamespaceStatusFound := newNamespaceMap.Status.CurrentMapping[localNs]
 		if !newRemoteNamespaceStatusFound || newRemoteNamespaceStatus.Phase != vkv1alpha1.MappingAccepted {
-			nh.stopNamespace(localNs, oldRemoteNamespaceStatus)
+			if newRemoteNamespaceStatusFound && newRemoteNamespaceStatus.Phase == vkv1alpha1.MappingCreationLoopBackOff {
+				nh.failingNamespace(localNs, oldRemoteNamespaceStatus)
+			} else {
```
[the remark this answers is not in the document]
Comment (giorio94) [reply pkg/virtualKubelet/reflection/namespacemap/namespace.go]: Applied the modification in the last commit. Let me know if that was what you were thinking about, then I can squash the commits.

[the remark this answers is not in the document]
Comment (aleoli) [reply pkg/virtualKubelet/reflection/namespacemap/namespace.go]: Totally fine for me 👍

**Comment by giorio94** [comment]
Closing in favor of #1131

**Pull request closed without merging**